consensus/e2e_tests/utils_test.go

```diff
@@ -45,6 +46,14 @@ const (
 
 type IdToNodeMapping map[typesCons.NodeId]*shared.Node
 
+// Used to mock state sync
+type StateSyncMockHelper struct {
+	maxheight uint64 // the max height of the node
+	minheight uint64 // the min height of the node
+}
+
+var stateSyncHelper StateSyncMockHelper
```

Comment:
Global variables are usually a really big code smell. I know it's "easy" and gets it working, but it becomes really hard to maintain & understand really fast.

Can you think of a way to make it more functional? How we pass everything around through variables.